apps/ftu/js/navigation.js

```diff
@@ -10,74 +11,87 @@ const DOGFOODSETTING = 'debug.performance_data.dogfooding';
 /*
   Steps of the First Time Usage App
 */
-var steps = {
-  1: {
+var steps = [
+  {
     onlyForward: true,
     hash: '#languages',
     requireSIM: false
   },
-  2: {
+  {
     onlyForward: false,
     hash: '#data_3g',
     requireSIM: true
   },
-  3: {
+  {
     onlyForward: false,
     hash: '#wifi',
     requireSIM: false
   },
-  4: {
+  {
     onlyForward: false,
     hash: '#date_and_time',
     requireSIM: false
   },
-  5: {
+  {
     onlyForward: false,
     hash: '#geolocation',
     requireSIM: false
   },
-  6: {
+  {
     onlyForward: false,
     hash: '#import_contacts',
     requireSIM: false
   },
-  7: {
+  {
     onlyForward: false,
     hash: '#firefox_accounts',
     requireSIM: false
   },
-  8: {
+  {
     onlyForward: false,
     hash: '#welcome_browser',
     requireSIM: false
   },
-  9: {
+  {
     onlyForward: false,
     hash: '#browser_privacy',
     requireSIM: false
   }
-};
-
-// Retrieve number of steps for navigation
-var numSteps = Object.keys(steps).length;
+];
 
-var Navigation = {
-  currentStep: 1,
-  previousStep: 1,
-  totalSteps: numSteps,
+exports.Navigation = {
+  stepCount: 0,
+  currentStepIndex: 0,
+  previousStepIndex: 0,
+  unskippedStepsCount: 0,
```

Comment:
using the occasion of the small rewriting, could make getters for the properties we don't want to be modified from outside? totally optional, but it feels like this is the perfect chance